coordinator/internal/authority/authority.go

```diff
@@ -23,24 +23,35 @@ import (
 	"github.com/google/go-sev-guest/validate"
 )
 
-type meshAuthority struct {
-	ca        *ca.CA
-	certs     map[string][]byte
-	certsMux  sync.RWMutex
-	manifests appendableList[*manifest.Manifest]
-	logger    *slog.Logger
+// Bundle is a set of PEM-encoded certificates for Contrast workloads.
+type Bundle struct {
+	Cert           []byte
+	MeshCA         []byte
+	IntermediateCA []byte
+	RootCA         []byte
 }
 
-func newMeshAuthority(ca *ca.CA, log *slog.Logger) *meshAuthority {
-	return &meshAuthority{
-		ca:        ca,
-		certs:     make(map[string][]byte),
+// Authority manages the manifest state of Contrast.
+type Authority struct {
+	ca         *ca.CA
+	bundles    map[string]Bundle
+	bundlesMux sync.RWMutex
+	manifests  appendableList[*manifest.Manifest]
+	logger     *slog.Logger
+}
+
+// New creates a new Authority instance.
+func New(caInstance *ca.CA, log *slog.Logger) *Authority {
+	return &Authority{
+		ca:        caInstance,
+		bundles:   make(map[string]Bundle),
 		manifests: new(appendable.Appendable[*manifest.Manifest]),
 		logger:    log.WithGroup("mesh-authority"),
 	}
 }
 
-func (m *meshAuthority) SNPValidateOpts(report *sevsnp.Report) (*validate.Options, error) {
+// SNPValidateOpts verifies that the report HOSTDATA field is registered in the active manifest.
```

Comment:
That's wrong, it returns any value for validation _beside HOSTDATA_, which is validated through the ValidateCallback.

Reply:
Well, it's not _wrong_: https://github.com/edgelesssys/contrast/blob/de5e0f93f2ba7f0f00ae1f2c79779dbf48a8b55a/coordinator/mesh.go#L94-L98.

But I agree that it does not capture all the function does and adjusted the docs.